Admin feature: adding multiple users from file.

JanCBrammer opened the pull request:
# Description

Currently, admins can only add one user at a time by submitting a form:
<img width="948" alt="add_single_user" src="https://user-images.githubusercontent.com/30125107/141997361-dc4f1816-02e0-4083-9624-8d00d0190f60.png">

This PR adds a feature that allows admins to add multiple users from a CSV file:
![add_multiple_users](https://user-images.githubusercontent.com/30125107/141997421-8f031266-50c0-4ed5-bfc8-ad5c8767080a.gif)

The feature has been requested by @alhoch.

# New dependencies
- [react-papaparse](https://react-papaparse.js.org/)

# Checklist
- [x] added feature to CHANGELOG
- [x] added acceptance test (passing locally and on CI)

PiTrem commented on spec/features/user_management_spec.rb:
```diff
@@ -0,0 +1,161 @@
+  end
+end
+
+PROCESSING_SUMMARY = # FIXME: string formatting might be brittle, i.e., matches might depend on runtime environment.
```
( squiggly heredoc ? )

JanCBrammer replied:
See 67a0899.

PiTrem commented on app/packs/src/admin/UserManagement.js:
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
+    for (let i = 0; i < this.newUsers.length; i++) {
+      this.newUsers[i].data.row = i + 1;
+    }
+    console.log(this.newUsers);
```
to be removed

JanCBrammer replied:
See d680d99.

PiTrem commented on app/packs/src/admin/UserManagement.js:
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
+  };
+
+  handleOnRemoveUserFile = (data) => {
+    console.log(data);
```
to be removed

JanCBrammer replied:
See d680d99.

PiTrem commented on app/packs/src/admin/UserManagement.js:
```diff
@@ -256,6 +259,143 @@ export default class UserManagement extends React.Component {
+      Promise.allSettled(promisedNewUsers)
+        .then((userResults) => {
+          this.setState({ createUserMessage: 'Finished processing user file.' });
+          this.showProcessingSummaryUserFile(userResults);
```
showProcessingSummaryUserFile is calling setState too. basically you have 2 setState in a row with each probably going to trigger rendering. (also not highly likely but the 2nd one could finish before the 1st one )
could you refactor using one  setState ?

JanCBrammer replied:
See e085070.

JanCBrammer commented:
@PiTrem, I addressed your comments above. If you're ok with the changes, I'll edit the commit history to be more comprehensive as discussed (basically one commit for the new feature, one commit for the tests, one commit for CHANGELOG).
